test/unit/TestLanguagePipeline.hs

```diff
@@ -254,14 +257,25 @@ testLanguagePipeline =
         "type annotations"
         [ testCase
             "annotate 1 + 1"
-            (assertEqual "foo"
+            (assertEqual "type annotations"
                (toListOf traverse (getSyntax [tmQ| 1 + 1 |]))
                (map (Forall []) [TyInt :->: TyInt :->: TyInt, TyInt, TyInt :->: TyInt, TyInt, TyInt])
             )
-        -- , testCase
-        --     "get all annotated variable types"
-        --     (let s = getSyntax
-        --            [tmQ| def f : (int -> int) -> int -> cmd unit = \g. \x. |]
+        , testCase
+            "get all annotated variable types"
+            (let s = getSyntax
+                   [tmQ| def f : (int -> int) -> int -> text = \g. \x. format (g x) end |]
+
+                 isVar (TVar {}) = True
+                 isVar _ = False
+                 getVars = map (_sTerm &&& _sType) . filter (isVar . _sTerm) . universe
```

Comment:
> `universe`

:exploding_head:

Reply:
See, I put that in there on purpose to give people ideas. :smile:  https://hackage.haskell.org/package/lens-5.2/docs/Control-Lens-Plated.html has some pretty powerful tools for working with ASTs.